### PR TITLE
Component selector has been created

### PR DESCRIPTION
**Pull request by iam-datsenko**
[DEMO LINK](https://iam-datsenko.github.io/react_goods-selector/)

### CODE REVIEW

**Review by iPanda177 — APPROVED** [review]
well done!

File: src/App.tsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import React from 'react';
+import { Component } from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
 
-export const goods = [
+export const goods: string[] = [
```
Comment (SerhiiKirik) [inline src/App.tsx:5]: extra
```suggestion
export const goods = [
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,71 +15,87 @@ export const goods = [
...
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
```
Comment (SerhiiKirik) [inline src/App.tsx:24]: why i can choose only 1?

File: src/App.tsx [hunk 3]
```diff
@@ -15,71 +15,87 @@ export const goods = [
...
+          <h1 className="title">No goods selected</h1>
+        )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map((good) => (
```
Comment (SerhiiKirik) [inline src/App.tsx:58]: can you split this into a separate component?

File: src/App.tsx [hunk 4]
```diff
@@ -15,71 +15,87 @@ export const goods = [
...
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+                      >
+                        +
+                      </button>
+                    )}
```
Comment (SerhiiKirik) [inline src/App.tsx:88]: you can create Button with own state)
Or put selected goods to the state & take info from this state)

**Review by SerhiiKirik — APPROVED** [review]
Good job 🚀 
But you can better, we always can better, so - let's learn 🎒

File: src/App.tsx [hunk 5]
```diff
@@ -15,71 +15,87 @@ export const goods = [
...
-        </tr>
+    return (
+      <main className="section container">
+        {selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```
[the remark this answers is not in the document]
Comment (iam-datsenko) [reply src/App.tsx:44]: I have an error without it - a control must be associated with a text label

[the remark this answers is not in the document]
Comment (SerhiiKirik) [reply src/App.tsx:44]: I believe, you can resolve this problem without disabling the linter)

**Comment by iam-datsenko** [comment]
> Good job 🚀 But you can better, we always can better, so - let's learn 🎒

**Pull request closed without merging**